db/hybrid_logical_vector.go

```diff
@@ -399,3 +406,39 @@ func (hlv *HybridLogicalVector) setPreviousVersion(source string, version uint64
 	}
 	hlv.PreviousVersions[source] = version
 }
+
+// toHistoryForHLV formats blip History property for V4 replication and above
+func (hlv *HybridLogicalVector) toHistoryForHLV() string {
+	// take pv and mv from hlv if defined and add to history
+	var pvString, mvString string
+	var s strings.Builder
+	// Merge versions must be defined first if they exist
+	if hlv.MergeVersions != nil {
+		mvString = extractStringFromMap(hlv.MergeVersions)
```

Comment:
Pushing this down into its own function that creates a new string is going to be allocation heavy.

We can either iterate over the map here and use the string builder directly, or pass the string builder down into a function if you want to keep it separate.